commands/deploy.go

```diff
@@ -212,7 +213,29 @@ Error: %s`, fprocessErr.Error())
 				Limits:   function.Limits,
 				Requests: function.Requests,
 			}
-			statusCode := proxy.DeployFunction(function.FProcess, services.Provider.GatewayURL, function.Name, function.Image, function.RegistryAuth, function.Language, deployFlags.replace, allEnvironment, services.Provider.Network, functionConstraints, deployFlags.update, deployFlags.secrets, allLabels, functionResourceRequest1)
+
+			var annotations map[string]string
+			if function.Annotations != nil {
+				annotations = *function.Annotations
+			}
+
+			statusCode := proxy.DeployFunction(
```

Comment:
This got a bit too long around the time we added "Language" and now it just got annotations plus the read-only flag. Probably time we refactored.. please could you raise an issue to refactor this into a `DeployFunctionStruct`?

Reply:
cc @LucasRoesler

Reply:
I agree that we should do this, I was actually thinking about this in my PR but didn't want to introduce a change like that at the same time as the read only behavior 👍

Reply:
@LucasRoesler agreed, it is best to work in increments. Please could you raise an issue for it today? 🐞

Reply:
done in #475